aten/src/ATen/native/RNN.cpp

```diff
@@ -365,23 +367,42 @@ struct FullBidirectionalLayer : Layer<Tensor, pair_of<dir_hidden_type>, pair_of<
 
   output_type operator()(const Tensor& input, const hidden_type& input_hidden, const param_type& params) const override {
     auto step_inputs = input.unbind(0);
-    auto fw_result = layer_(step_inputs, input_hidden.first, params.first);
-    auto fw_output = at::stack(fw_result.outputs, 0);
 
-    auto rev_step_inputs = reverse(std::move(step_inputs));
-    auto rev_result = layer_(rev_step_inputs, input_hidden.second, params.second);
-    std::reverse(rev_result.outputs.begin(), rev_result.outputs.end());
+    using unstacked_output_type = typename FullLayer<dir_hidden_type, cell_params>::unstacked_output_type;
+
+#if AT_PARALLEL_NATIVE_TBB
+    unstacked_output_type fw_result;
+    at::Tensor fw_output;
+
+    unstacked_output_type rev_result;
+    at::Tensor rev_output;
+
+    at::intraop_invoke([&]() {
+      fw_result = layer_(step_inputs, input_hidden.first, params.first);
+      fw_output = at::stack(fw_result.outputs, 0);
+    }, [&]() {
+      rev_result = layer_(step_inputs, input_hidden.second, params.second, /* reverse */ true);
+      rev_output = at::stack(rev_result.outputs, 0);
+    });
+#else
+    unstacked_output_type fw_result;
+    at::Tensor fw_output;
+    auto fut = at::intraop_launch_future([&]() {
```

Comment:
Is this actually a good idea for something like openmp? Won't it execute two OpenMP operators concurrently which might oversubscribe the system?

Reply:
in case of openmp this will be inline, oversubscribing depends on the execution environment - in some cases we can benefit form extra parallelizm, but not always, depending on whether there're other inter-op tasks, how many parallel requests we handle, are there other processes running, machine capability, this needs to be benchmarked each time